tasks/eth/010-2-sc-changes/SignFromJson.s.sol

```diff
@@ -0,0 +1,330 @@
+import {OptimismPortal} from "@eth-optimism-bedrock/src/L1/OptimismPortal.sol";
+import {L1CrossDomainMessenger} from "@eth-optimism-bedrock/src/L1/L1CrossDomainMessenger.sol";
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {ISemver} from "@eth-optimism-bedrock/src/universal/ISemver.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {EIP1967Helper} from "@eth-optimism-bedrock/test/mocks/EIP1967Helper.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import {GnosisSafe} from "safe-contracts/GnosisSafe.sol";
+import {ModuleManager} from "safe-contracts/base/ModuleManager.sol";
+
+// Interfaces of the new contracts we need to check.
+interface ILivenessModuleFetcher {
+    function fallbackOwner() external view returns (address fallbackOwner_);
+    function getRequiredThreshold(uint256 _numOwners) external view returns (uint256 threshold_);
+    function livenessGuard() external view returns (address livenessGuard_);
+    function livenessInterval() external view returns (uint256 livenessInterval_);
+    function minOwners() external view returns (uint256 minOwners_);
+    function safe() external view returns (address safe_);
+    function thresholdPercentage() external view returns (uint256 thresholdPercentage_);
+    function version() external view returns (string memory);
+}
+
+interface ILivenessGuardFetcher {
+    function lastLive(address) external view returns (uint256);
+    function safe() external view returns (address safe_);
+    function version() external view returns (string memory);
+}
+
+interface IDeputyGuardianModuleFetcher {
+    function deputyGuardian() external view returns (address deputyGuardian_);
+    function safe() external view returns (address safe_);
+    function superchainConfig() external view returns (address superchainConfig_);
+    function version() external view returns (string memory);
+}
+
+contract SignFromJson is OriginalSignFromJson {
+    using LibString for string;
+
+    address constant SENTINEL_MODULE = address(0x1);
+    bytes32 constant GUARD_STORAGE_SLOT = 0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8;
+    bytes32 constant GUARDIAN_SLOT = bytes32(uint256(keccak256("superchainConfig.guardian")) - 1);
+
+    // Chains for this task.
+    string constant l1ChainName = "mainnet";
+    string constant l2ChainName = "op";
+
+    // Properties we want to verify.
+    uint256 constant expectedGuardLivenessInterval = 14 weeks;
+    uint256 constant expectedGuardMinOwners = 8;
+    uint256 constant expectedGuardThresholdPercentage = 75;
+    uint256 constant expectedThreshold = 10;
```

Comment:
It would be good to add an `expectedOwnersCount` and assert that the owners are unchanged.

Reply:
Done in 74ed190